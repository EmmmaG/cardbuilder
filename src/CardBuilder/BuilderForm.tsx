import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

import TextInput from '../Components/TextInput'
import Button from '../Components/Button'
import Divider from '../Components/Divider'


const styles = (theme: Theme) => ({
    title: {
        fontSize: '32px',
        fontWeight: 900,
        color: '#374D60',
        marginLeft: '10px',
        textAlign: 'left' as 'left',
    },
    mainContainer: {
        backgroundColor: 'white',
        maxWidth: '430px',
        textAlign: 'center' as 'center',
        padding: '30px',
        display: 'inline-block',
    },
    buttonContainer: {
        marginTop: '30px',
    },
})

export interface FormValue {
    givenName: string
    surname: string
    email: string
    phone: string
    houseName: string
    street: string
    suburb: string
    state: string
    postcode: string
    country: string
}

export interface Props extends WithStyles<typeof styles> {
    onFormChange: (formValue: FormValue) => void
}

export interface State {
    formValue: FormValue
}

export const emptyFormValue: FormValue = {
    givenName: '',
    surname: '',
    email: '',
    phone: '',
    houseName: '',
    street: '',
    suburb: '',
    state: '',
    postcode: '',
    country: '',
}

type FormKey = keyof FormValue

export class BuilderForm extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            formValue: emptyFormValue,
        }
    }

    private onTextFieldChange = (value: string, formKey: FormKey) => {
        const currentFormValue = this.state.formValue

        this.setState({
            formValue: {
                ...currentFormValue,
                [formKey]: value,
            }
        }, () => {this.props.onFormChange(this.state.formValue)})
    }

    private renderTextInput = (title: string, formKey: FormKey) => {
        return (
            <TextInput key={formKey} title={title} onChange={(value) => this.onTextFieldChange(value, formKey)} />
        )
    }

    private renderPersonalDetails = () => {
        const nameRow = [{
            title: 'GIVEN NAME',
            formKey: 'givenName' as FormKey,
        }, {
            title: 'SURNAME',
            formKey: 'surname' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const detailRow = [{
            title: 'EMAIL',
            formKey: 'email' as FormKey,
        }, {
            title: 'PHONE',
            formKey: 'phone' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        return (
            <div>
                <Divider title='PERSONAL DETAILS' />
                <div>
                    {nameRow}
                </div>
                <div>
                    {detailRow}
                </div>
            </div>
        )
    }

    private renderAddress = () => {
        const streetRow = [{
            title: 'HOUSE NAME OR #',
            formKey: 'houseName' as FormKey,
        }, {
            title: 'STREET',
            formKey: 'street' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const suburbRow = [{
            title: 'SUBURB',
            formKey: 'suburb' as FormKey,
        }, {
            title: 'STATE',
            formKey: 'state' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        const postcodeRow = [{
            title: 'POSTCODE',
            formKey: 'postcode' as FormKey,
        }, {
            title: 'COUNTRY',
            formKey: 'country' as FormKey,
        }].map(element => this.renderTextInput(element.title, element.formKey))

        return (
            <div>
                <Divider title='PERSONAL DETAILS' />
                <div>
                    {streetRow}
                </div>
                <div>
                    {suburbRow}
                </div>
                <div>
                    {postcodeRow}
                </div>
            </div>
        )
    }

    private renderButtons = () => (
        <div className={this.props.classes.buttonContainer}>
            <Button
                buttonContent='Upload Avatar'
                onClick={() => console.log('a')}
                backgroundColor='#768E9C'
            />
            <Button
                buttonContent='Create hCard'
                onClick={() => console.log('a')}
                backgroundColor='#43A8E0'
            />
        </div>
    )

    public render() {
        const { classes } = this.props

        return (
            <div className={classes.mainContainer}>
				<Typography className={classes.title}>hCard Builder</Typography>

                {this.renderPersonalDetails()}

                {this.renderAddress()}

                {this.renderButtons()}
            </div>
        )
    }
}

export default withStyles(styles)(BuilderForm)
