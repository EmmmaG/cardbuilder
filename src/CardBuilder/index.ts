export { default } from './CardBuilder'
export * from './CardBuilder'
