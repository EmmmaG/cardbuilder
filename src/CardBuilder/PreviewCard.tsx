import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

import { FormValue } from './BuilderForm'
import person from '../assets/person.png'

import CardRow from '../Components/CardRow'

const styles = (theme: Theme) => ({
    mainContainer: {
        width: '450px',
        display: 'inline-block',
    },
    title: {
        textAlign: 'right' as 'right',
        color: theme.palette.grey[400],
        fontSize: '18px',
        fontWeight: 500,
        marginTop: '5px',
        marginBottom: '5px',
    },
    cardContainer: {
        position: 'relative' as 'relative',
        width: '100%',
        boxShadow: `0.5px 2px ${theme.palette.grey[300]}`,
    },
    nameContainer: {
        height: '110px',
        backgroundColor: '#394F62',
    },
    name: {
        color: 'white',
        fontFamily: '"Big Caslon", "Times New Roman", "serif"',
        fontSize: '26px',
        fontWeight: 600,
        paddingTop: '55px',
        paddingLeft: '20px',
    },
    profile: {
        position: 'absolute' as 'absolute',
        zIndex: 1,
        top: '15px',
        right: '15px',
        height: '110px',
        width: '85px',
        backgroundColor: 'white',
        padding: '5px',
        border: `1px solid ${theme.palette.grey[300]}`,
        boxShadow: `1px 2px ${theme.palette.grey[300]}`,
    },
    contentsContainer: {
        padding: '20px 30px',
        backgroundColor: 'white',

    },
    postCountryContainer: {
        display: 'flex',
        justifyContent: 'space-between',
    },
})

export interface Props extends WithStyles<typeof styles> {
    contents: FormValue
    avatar?: string
}

export class PreviewCard extends React.Component<Props > {
    private renderCardContents = () => {
        const { contents, classes } = this.props
        const suburbRow = `${contents.suburb}${(contents.suburb && contents.state) ? ',' : ''} ${contents.state}`

        return (
            <div className={classes.contentsContainer}>
                <CardRow title='EMAIL' content={contents.email} />
                <CardRow title='PHONE' content={contents.phone} />
                <CardRow title='ADDRESS' content={`${contents.houseName} ${contents.street}`} />
                <CardRow content={suburbRow} />
                <div className={classes.postCountryContainer}>
                    <CardRow title='POSTCODE' content={contents.postcode} inline />
                    <CardRow title='COUNTRY' content={contents.country} inline />
                </div>
            </div>
        )
    }

    public render() {
        const { classes, contents, avatar } = this.props

        return (
            <div className={classes.mainContainer}>
                <Typography className={classes.title}>HCARD PREVIEW</Typography>
                <div className={classes.cardContainer}>
                    <div className={classes.nameContainer}>
                        <Typography className={classes.name}>{`${contents.givenName} ${contents.surname}`}</Typography>
                    </div>
                    <img src={avatar || person} className={classes.profile}/>
                    {this.renderCardContents()}
                </div>
            </div>
        )
    }
}

export default withStyles(styles)(PreviewCard)
