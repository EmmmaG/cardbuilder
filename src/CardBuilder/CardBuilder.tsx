import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import BuilderForm, { FormValue, emptyFormValue } from './BuilderForm'
import PreviewCard from './PreviewCard'

const styles = (theme: Theme) => ({
    builderContainer: {
        display: 'flex',
        flexDirection: 'row' as 'row',
        alignItems: 'center',
        justifyContent: 'space-evenly',
        flexWrap: 'wrap' as 'wrap',
    }
})

export interface Props extends WithStyles<typeof styles> {
}

export interface State {
    cardContents: FormValue
}

export class CardBuilder extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            cardContents: emptyFormValue,
        }
    }

    private setCardContents = (cardContents: FormValue) => {
        this.setState({ cardContents })
    }

    public render() {
        const { classes } = this.props
        const { cardContents } = this.state

        return (
            <div className={classes.builderContainer}>
                <BuilderForm onFormChange={this.setCardContents} />
                <PreviewCard contents={cardContents} />
            </div>
        )
    }
}

export default withStyles(styles)(CardBuilder)
