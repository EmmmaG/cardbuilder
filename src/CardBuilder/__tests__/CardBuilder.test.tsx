import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import CardBuilder, { Props } from '../CardBuilder'

const props: Props = {
	classes: idObj,
}

describe('CardBuilder', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<CardBuilder {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
    })
})
