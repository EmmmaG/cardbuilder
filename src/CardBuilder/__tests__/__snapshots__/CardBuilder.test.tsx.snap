// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBuilder renders correctly 1`] = `
<div
  className="CardBuilder-builderContainer-1"
>
  <WithStyles(BuilderForm)
    onAvatarSelected={[Function]}
    onFormChange={[Function]}
  />
  <WithStyles(PreviewCard)
    avatar=""
    contents={
      Object {
        "country": "",
        "email": "",
        "givenName": "",
        "houseName": "",
        "phone": "",
        "postcode": "",
        "state": "",
        "street": "",
        "suburb": "",
        "surname": "",
      }
    }
  />
</div>
`;
